Modules/socketmodule.c

```diff
@@ -3520,7 +3521,7 @@ sock_connect_ex(PySocketSockObject *s, PyObject *addro)
     }
 
     res = internal_connect(s, SAS2SA(&addrbuf), addrlen, 0);
-    if (res < 0)
+    if (res == -1 && PyErr_Occurred())
         return NULL;
```

Comment:
And here you can do:
```suggestion
    if (res == -1 && PyErr_Occurred()) {
        return NULL;
    }
```
(This scheme is discouraged nowadays, but refactoring `internal_connect` to use something else is probably not worth it.)